problems/[230302] 101. Symmetric Tree.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/symmetric-tree
+ *
+ * Definition for a binary tree node.
+ * function TreeNode(val, left, right) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.left = (left===undefined ? null : left)
+ *     this.right = (right===undefined ? null : right)
+ * }
+ */
+/**
+ * @param {TreeNode} root
+ * @return {boolean}
+ */
+const isSymmetric = function (root) {
+  const leftTreeValues = [];
+
+  (function leftTreeDfs(tree) {
+    if (!tree) {
+      leftTreeValues.push(101);
+
+      return;
+    }
+
+    leftTreeValues.push(tree.val);
+
+    if (tree.left || tree.right) {
+      leftTreeDfs(tree.left);
+      leftTreeDfs(tree.right);
+    }
+  })(root.left);
+
+  function compareRightToTheLeftTreeItems(tree, LeftItemsArr) {
+    const compareLeftValue = LeftItemsArr.pop();
+
+    if (!tree) {
+      if (compareLeftValue !== 101) {
+        LeftItemsArr.push(101);
+      }
+
+      return;
+    }
+
+    if (compareLeftValue !== tree.val) {
+      LeftItemsArr.push(101);
+
+      return;
+    }
+
+    if (tree.left || tree.right) {
+      compareRightToTheLeftTreeItems(tree.right, LeftItemsArr);
+      compareRightToTheLeftTreeItems(tree.left, LeftItemsArr);
+    }
+  }
+
+  compareRightToTheLeftTreeItems(root.right, leftTreeValues.reverse());
+
+  return leftTreeValues.length ? false : true;
```

Comment:
```suggestion
  return leftTreeValues.length === 0;
```

삼항연산자는 truthy 혹은 falsy 값을 분기해 특정 표현식을 반환하게 하는 연산자인데, 단순히 true 혹은 false 값으로 나타내는 거라면 !!연산자 혹은 !연산자로 만들어줘도 될 것 같아요

Reply:
오호 👍